docs/routing/introduction.md

```diff
@@ -54,54 +56,82 @@ function Home() {
           <a>About Us</a>
         </Link>
       </li>
+      <li>
+        <Link href="/blog/hello-world">
+          <a>Blog Post</a>
+        </Link>
+      </li>
     </ul>
   )
 }
 
 export default Home
 ```
 
-When linking to a route with [dynamic path segments](/docs/routing/dynamic-routes.md) you have to provide `href` and `as` to make sure the router knows which JavaScript file to load.
+In the example above we have multiple links, each one maps a path (`href`) to a known page:
+
+- `/` → `pages/index.js`
+- `/about` → `pages/about.js`
+- `/blog/hello-world` → `pages/blog/[slug].js`
+
+### Linking to dynamic paths
 
-- `href` - The name of the page in the `pages` directory. For example `/blog/[slug]`.
-- `as` - The url that will be shown in the browser. For example `/blog/hello-world`.
+You can also use interpolation to create the path, which comes in handy for [dynamic route segments](#dynamic-route-segments). For example, to show a list of posts which have been passed to the component as a prop:
 
 ```jsx
 import Link from 'next/link'
 
-function Home() {
+function Posts({ posts }) {
   return (
     <ul>
-      <li>
-        <Link href="/blog/[slug]" as="/blog/hello-world">
-          <a>To Hello World Blog post</a>
-        </Link>
-      </li>
+      {posts.map((post) => (
+        <li key={post.id}>
+          <Link href={`/blog/${encodeURIComponent(post.slug)}`}>
+            <a>{post.title}</a>
+          </Link>
+        </li>
+      ))}
     </ul>
   )
 }
 
-export default Home
+export default Posts
 ```
 
-The `as` prop can also be generated dynamically. For example, to show a list of posts which have been passed to the page as a prop:
+> [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) is used in the example to keep the path utf-8 compatible.
+
+Alternatively, using an URL Object:
 
 ```jsx
-function Home({ posts }) {
+import Link from 'next/link'
+
+function Posts({ posts }) {
   return (
     <ul>
       {posts.map((post) => (
         <li key={post.id}>
-          <Link href="/blog/[slug]" as={`/blog/${post.slug}`}>
+          <Link
+            href={{
+              pathname: '/blog/[slug]',
+              query: { slug: post.slug },
```

Comment:
This doesn't correctly work as expected and will need updating in the router before we recommend this

Reply:
@ijjk What should we do for now?

Reply:
We'll need to remove this from the docs for now and wait to add it in a follow-up until proper support is landed. I opened an issue to track this [here](https://github.com/vercel/next.js/issues/16771)

Reply:
@ijjk I'm going to resolve this as you did some PRs to fix this. Lmk if this is still an issue.